plugin.video.iribtv/default.py

```diff
@@ -0,0 +1,23 @@
+import xbmc, xbmcgui, os, xbmcaddon, urllib2
+from BeautifulSoup import BeautifulSoup
+
+__settings__ = xbmcaddon.Addon()
+home = __settings__.getAddonInfo('path')
+addonname = __settings__.getAddonInfo('name')
+icon = xbmc.translatePath(os.path.join(home, 'icon.png'))
+li = xbmcgui.ListItem ('IRIB TV3')
+li.setArt({ 'thumb': icon })
+li.setInfo('video', {'plot': 'Live Stream'})
+
+livePageUrl = "https://www.tv3.ir/live"
+
+request = urllib2.Request(livePageUrl, headers={'User-Agent' : "Kodi"})
+
+response = urllib2.urlopen(request)
+html = response.read()
+
+soup = BeautifulSoup(html)
+stream = soup.find('video', preload='auto')
+
+if stream is not None and stream['data-source'] is not None:
+        xbmc.Player().play(stream['data-source'], li, False)
```

Comment:
Calling `xbmc.Player().play(...)` isn't allowed for a plugin.
You have to create a directory listing in your root path of the add-on.